lib/galaxy/config_watchers.py

```diff
@@ -37,6 +39,12 @@ def __init__(self, app, start_thread=True):
             self.job_rule_watcher = get_watcher(app.config, 'watch_job_rules', monitor_what_str='job rules')
         else:
             self.job_rule_watcher = get_watcher(app.config, '__invalid__')
+        self.core_config_watcher = get_watcher(
+            app.config,
+            'watch_core_config',
+            monitor_what_str='core config file',
+            event_handler_class=SimpleFileModifiedEventHandler
```

Comment:
```suggestion
```
As mentioned below, the handler doesn't work for moving files into place, but the default EventHandler works fine.
Plus it makes sure that there really are changes that require a reload by checking the md5 of the file.

Reply:
Fixed.